catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/monitor/email/SubscribedWorkspaceNotifierTest.java

```diff
@@ -0,0 +1,163 @@
+import ddf.catalog.data.Result;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.AttributeBuilder;
+import ddf.catalog.filter.ExpressionBuilder;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.security.Subject;
+import ddf.security.SubjectIdentity;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.shiro.util.ThreadContext;
+import org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceConstants;
+import org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceMetacardImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.opengis.filter.And;
+import org.opengis.filter.Filter;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SubscribedWorkspaceNotifierTest {
+
+  private SubscribedWorkspaceNotifier subscribedWorkspaceNotifier;
+
+  @Mock private EmailNotifier mockEmailNotifier;
+
+  @Mock private QueryResponse mockQueryResponse;
+
+  @Mock private CatalogFramework mockCatalogFramework;
+
+  @Mock private SubjectIdentity mockSubjectIdentity;
+
+  @Mock private QueryRequest mockQueryRequest;
+
+  @Mock private Attribute mockAttribute;
+
+  @Before
+  public void setup()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    WorkspaceMetacardImpl mockWorkspaceMetacard = mock(WorkspaceMetacardImpl.class);
+    QueryResponse mockWorkspaceMetacardQueryResponse = mock(QueryResponse.class);
+    ExpressionBuilder mockExpressionBuilder = mock(ExpressionBuilder.class);
+    AttributeBuilder mockAttributeBuilder = mock(AttributeBuilder.class);
+    FilterBuilder mockFilterBuilder = mock(FilterBuilder.class);
+    Subject mockSubject = mock(Subject.class);
+    Result mockResult = mock(Result.class);
+
+    ThreadContext.bind(mockSubject);
+    List<Result> resultsForMockQueryResponse = Collections.singletonList(mockResult);
+    Map<String, Serializable> propertiesForMockQueryRequest = new HashMap<>();
+    propertiesForMockQueryRequest.put("id", "1234abcd");
+    List<Serializable> valuesForMockAttribute = Collections.singletonList("test@email.com");
+    Set<String> metacardTagsForMockWorkspaceMetacard =
+        Collections.singleton(WorkspaceConstants.WORKSPACE_TAG);
+
+    when(mockFilterBuilder.allOf(any(Filter.class), any(Filter.class))).thenReturn(mock(And.class));
+    when(mockFilterBuilder.attribute(anyString())).thenReturn(mockAttributeBuilder);
+    when(mockAttributeBuilder.is()).thenReturn(mockExpressionBuilder);
```

Comment:
Do you need to mock both parameters to the `allOf` call if you are also mocking the return of the `allOf` call?

Reply:
Yes, I tried to only mock the `allOf` call and was getting a `NullPointerException` when it hit the `filterBuilder.attribute` calls

Reply:
You can use the GeoToolsFilterBuilder in tests instead of mocking the filterbuilder; this makes the code a little cleaner

Reply:
Yeah i've recommended using GeoToolsFilterBuilder before since it cleans it up so much. Though always conflicted because I remember being told not to use other implementations in tests 😂  Could be a topic for a Test CoP when they start up again.